packages/devextreme/js/ui/date_box/ui.date_box.mask.js

```diff
@@ -306,6 +306,8 @@ const DateBoxMask = DateBoxBase.inherit({
     _searchString(char) {
         if(!isNaN(parseInt(this._getActivePartProp('text')))) {
             return;
+        } else {
+            this._toggleAmPmOnChar(char.toLowerCase(), this._getActivePartProp('text'));
```

Comment:
if we toggled am/pm, we may return from the function, i suppose, as you did before